drink/src/lib.rs

```diff
@@ -77,7 +96,65 @@ impl<R: Runtime> Sandbox<R> {
     ///
     /// By default, a new `Sandbox` instance is created with a noop debug extension. This method
     /// allows to override it with a custom debug extension.
-    pub fn override_debug_handle(&mut self, d: DebugExt) {
+    pub fn override_debug_handle(&mut self, d: TracingExt) {
         self.externalities.register_extension(d);
     }
+
+    /// Registers the extension for intercepting calls to contracts.
+    fn setup_mock_extension(&mut self) {
+        self.externalities
+            .register_extension(InterceptingExt(Box::new(MockingExtension {
+                mock_registry: Arc::clone(&self.mock_registry),
+            })));
+    }
+}
+
+/// Runtime extension enabling contract call interception.
+struct MockingExtension<AccountId: Ord> {
+    /// Mock registry, shared with the sandbox.
+    ///
+    /// Potentially the runtime is executed in parallel and thus we need to wrap the registry in
+    /// `Arc<Mutex>` instead of `Rc<RefCell>`.
+    mock_registry: Arc<Mutex<MockRegistry<AccountId>>>,
+}
+
+impl<AccountId: Ord + Decode> InterceptingExtT for MockingExtension<AccountId> {
+    fn intercept_call(
+        &self,
+        contract_address: Vec<u8>,
+        _is_call: bool,
+        input_data: Vec<u8>,
+    ) -> Vec<u8> {
+        let contract_address = Decode::decode(&mut &contract_address[..])
```

Comment:
Would `contract_address.as_mut_slice()` work? https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_mut_slice

Reply:
nope:
```
the trait `Input` is not implemented for `[u8]`
```

that's awful, I agree :P